src/main/java/org/testng/IAlterTestName.java

```diff
@@ -0,0 +1,19 @@
+package org.testng;
+
+/**
+ * This interface lets you alter the test name of an {@link ITestResult} object.
+ * Sample : <br>
+ *
+ * <pre>
+ *     if (testResult instanceOf IAlterTestName) {
+ *         ((ITestResult) testResult).setTestName(newName);
+ *     }
+ *
+ * </pre>
+ */
+public interface IAlterTestName {
```

Comment:
The reason why I felt we should use a new interface is because 
* I needed a new method signature to be introduced which users could use to alter the name, but i cannot add it into the existing `ITestResult` because that can break compatibility to users, incase someone has their own custom implementation of this interface(I dont know if that's a fair assumption)
* I couldnt add this directly to the implementation viz., `TestResult` class because that would basically expose/tie the callers to the underlying implementation. So the whole point of runtime polymorphism is lost.

Considering both these two, I felt that this was perhaps an acceptable approach.